jupyterhub_singleuser_profiles/ui/src/SizesForm/SizesForm.tsx

```diff
@@ -69,17 +64,17 @@ const SizesForm: React.FC<ImageFormProps> = ({ uiConfig }) => {
   }, [sizeList]);
 
   React.useEffect(() => {
-    if (!sizeList) {
-      return;
-    }
-
-    for (let i = 0; i < sizeList.length; i++) {
-      if (sizeList[i] === selectedSize || selectedSize === 'Default') {
-        return;
+    if (userConfig && sizeDescriptions) {
+      const description = sizeDescriptions.find((sd) => sd.name === userConfig.last_selected_size);
```

Comment:
`find` can return undefined, if you're comfortable with it leave it be but since the value can be destructured... similar comment about optional chains or destructuring goes here

Reply:
resolved (i don't have perms to close this comment)